web3.js/src/transaction.ts

```diff
@@ -172,8 +191,22 @@ export class Transaction {
   /**
    * Construct an empty Transaction
    */
-  constructor(opts?: TransactionCtorFields) {
-    opts && Object.assign(this, opts);
+  constructor(opts?: TransactionCtorFields | TransactionCtorFields_DEPRECATED) {
```

Comment:
Still needs deprecation using TypeScript method overloads.

Reply:
Stella was working on this, so I'll check in with her to see how it's going.

Reply:
https://github.com/solana-labs/solana/pull/24211#discussion_r854668811

Reply:
Solved it! My initial take was way too complicated. ([link](https://www.typescriptlang.org/play?#code/C4TwDgpgBAQgNgewMYGsAWBDAzmqBeKLYAJwEsA7AcwG4AoW0SKAFWI3KwyWFIXIGFgCYgDFSEOABMsAfQAiAUQAKAJQX8AgswVz8UAPIBbUsAA8rdp268BQ0eKlYANFADkcDMAhF4ydNjRXAD4oADIoAG9aKChiCCQIcmBfVEwcAH4ALlhEVICoAB8ocgBXODg6AF86BnBoCw4uHj5BYTEJaT0omI8vH1z-DOzumNGAIwG0tGyUwbRqKGjRnuxgADUMOFJJWYAJCFJKNGBs0sMxiGIl6qWEYDRLgGVgEoAzV6zCEgoaWhvaJAeLBYKD8PjkeLNciRJYyIHrTbbPYHI4nYolc6XQpQErkSQQV4UCCSOgxGRxBJJWZTGaTfJFUrlbG4-GEiEk+gxAD0XKgcgJRKg92gFC8xFeXGgr2EQoeUFeuOsfCFdSgAHcTLgMFAtkQoAhXvqAG6XRAYaRLJZcgBU1qgAAF8WAKZ5iUsoHaAAYUxLJOk4T1QNJQC6JKBOl1eSQAOnddqUcAg2GgYGwIM9vW8fr8NMiEz8uwCtILARc8I2Wx2A32h2OpwxF2IlUDFCISZjcZYaFIII1TIu4YgzvirskUAoUG1AA1owAGOdQE3ELA2fWG+1YBAecgYLlqiBjADM0YAVlhY6NrVylkg+ERiCVuMIABQIMDALCfFhsRpK2xtBxpHkZQ1E0bQ5AASlJRZuV5Zg5QhNVCEOHcXjiG87xIR87Ffd9P2yBorChVp7A6LAoM5KAeS7MNWUFYVx0MMBE0MX1PFXFdKFQkp0JiW8OCwp9iFwj9PkIpobBI9pHGxcS-ykwDZEUVR1C0HQIJhZYYlIQ1nwAQjfD8NJGLTRjiNDyGg5ZKndUYdKgETzzSfQ1XIJRiDfS5QGfVwfSpf1Agg4zbOWaiAElgFcEEGK3SQAFoiBARN9XfGxzxC0Z+M3RNozVDBiHIHz+DidiqEnKBPTBcgIT-QMNXuCq-OzPIA3HEEIxHKNo261wKNMvjMP1KR9DwvRDJBbBv0sCSWjsaSgOU0C1LkKytPuHto3JeJfWpfICFikaP2jJrdpwVaYkqKAJCwaATK08LIuihCICQxLksMtKL36rLgGKF7DpBfbRsmuTiLmxTztGdbzzhVYKyRasUWOPREIB6NM36HMAnSdG4cRKsCyR4BIZiaHNpOgKUf+vDcb6Zq5hx-MWvmDKbNM64-noH6hskKqaqhKmkL5yEbGfCIKaxnBslcDAxiQVxKgo7nEOFv9BdBcERb4MWMfp3MIiZuZpdl+WyzxytkVrNEAEZZztypFeoIA))

Reply:
Stella pushed some changes for this. I also added a separate constructor overload for `nonce` transactions to address your [comment](https://github.com/solana-labs/solana/issues/23949#issuecomment-1092221677) about transactions being constructed with either `nonce` or `latestBlockhash`. It seemed to be the cleanest way to do it. What are your thoughts?